### DISCUSSION TITLE
FR: Metadata "wrangling" functionality

### DISCUSSION THREAD

**Discussion by arminta7** [mdelobelle/metadatamenu#152] [Ideas]
I run into trouble sometimes keeping track of what fields I've used in my vault, making sure they're all worded the same way, etc. So just like the tag wrangler plugin, I would love to see some metadata wrangling functionality built in. To see what fields I'm using, merge fields, rename fields, see the values I've used for each field, etc.

**Comment by mdelobelle** [comment]
that sounds really doable.

As of today, all fields definition  are stored in à index. If you are curious you can access it through the console : app.plugins.plugins["metadata-menu"].fieldIndex

So,as you acn see, all the data is already available

Could you describe a little bit more the way you'd like fields to be displayed?
and what you'd like to do with them?

**Comment by arminta7** [comment]
Oh yes, I do see the data now!

I'd like to see all of the fields I've used in my vault, both defined within Metadata Menu (fileclasses, global, etc) and otherwise used. How many times they are used. All of the values I have used for each specific field, whether defined by Metadata Menu or not. What fileClasses the fields are used in. 

Once I can see that info, being able to rename and merge fields throughout the whole vault (like tag wrangler). 

Maybe the initial view would just be a long list of fields with the number of times they are used (like the tag pane), then you could click on a specific field to see where it's used, what values have been used, if it's in Metadata menu or not. Maybe right click from initial view to rename. Any sort of bulk action functionality that can be added is great, maybe even bulk renaming field values. 

I have thousands of pages in my vault (and sometimes hundreds of metadata fields in a page) and haven't always been great about keeping consistent naming for fields and their values, so I need to go through and see what doesn't match up and change everything, then keep them consistent!

**Comment by mdelobelle** [comment]
Well this is a little bit more complex 😅
The need to report all fields with their usage could be covered with a dataview table and some js iteration on fieldIndex. I'll move this issue in the discussion section and maybe we can try to build a dataview snippet to deal with it?

For the rename fields part, I think this would be dangerous in term of data integrity.
This would indeed be based on a big search and replace on field names. Unlike the tags, the naming of inline fields is not a core principle of Obsidian at the moment, so we could tamper with the users data in a way that he doesn't want to.

On top of that, this would be heavy in terms of performance: each field renaming would lead into a cache resolution, leading to lookup fields resolution and so on...

For the merging part, this is what the `extend` attribute is made for. Imagine you have #A and #B and you want to merge them (have their fields grouped), you can create a #C with `extends: [A, B]`. Could this suit part of your need?

Many of your ideas are great. I certainly can develop some of them. We have to break it down in smaller pieces 🙂
